go/cmd/dolt/commands/sqlserver/serverconfig.go

```diff
@@ -25,140 +27,221 @@ type LogLevel string
+func (cfg *commandLineServerConfig) withPort(port int) *commandLineServerConfig {
+	cfg.port = port
+	return cfg
+}
+
+// withUser updates the user and returns the called `*commandLineServerConfig`, which is useful for chaining calls.
+func (cfg *commandLineServerConfig) withUser(user string) *commandLineServerConfig {
+	cfg.user = user
+	return cfg
+}
+
+// withPassword updates the password and returns the called `*commandLineServerConfig`, which is useful for chaining calls.
+func (cfg *commandLineServerConfig) withPassword(password string) *commandLineServerConfig {
+	cfg.password = password
+	return cfg
+}
+
+// withTimeout updates the timeout and returns the called `*commandLineServerConfig`, which is useful for chaining calls.
+func (cfg *commandLineServerConfig) withTimeout(timeout uint64) *commandLineServerConfig {
+	cfg.timeout = timeout
+	return cfg
+}
+
+// withReadOnly updates the read only flag and returns the called `*commandLineServerConfig`, which is useful for chaining calls.
+func (cfg *commandLineServerConfig) withReadOnly(readonly bool) *commandLineServerConfig {
+	cfg.readOnly = readonly
+	return cfg
+}
+
+// withLogLevel updates the log level and returns the called `*commandLineServerConfig`, which is useful for chaining calls.
+func (cfg *commandLineServerConfig) withLogLevel(loglevel LogLevel) *commandLineServerConfig {
+	cfg.logLevel = loglevel
+	return cfg
+}
+
+func (cfg *commandLineServerConfig) withDBNamesAndPaths(dbNamesAndPaths []env.EnvNameAndPath) *commandLineServerConfig {
+	cfg.dbNamesAndPaths = dbNamesAndPaths
+	return cfg
+}
+
+// DefaultServerConfig creates a `*ServerConfig` that has all of the options set to their default values.
+func DefaultServerConfig() *commandLineServerConfig {
+	return &commandLineServerConfig{
+		host:           "localhost",
+		port:           3306,
+		user:           "root",
+		password:       "",
+		timeout:        30 * 1000,
+		readOnly:       false,
+		logLevel:       LogLevel_Info,
+		autoCommit:     true,
+		maxConnections: 0,
+	}
+}
+
+// Validate returns an `error` if any field is not valid.
+func ValidateConfig(config ServerConfig) error {
+	if config.Host() != "localhost" {
+		ip := net.ParseIP(config.Host())
+		if ip == nil {
+			return fmt.Errorf("address is not a valid IP: %v", config.Host())
+		}
 	}
+	if config.Port() < 1024 || config.Port() > 65535 {
+		return fmt.Errorf("port is not in the range between 1024-65535: %v\n", config.Port())
+	}
+	if len(config.User()) == 0 {
+		return fmt.Errorf("user cannot be empty")
+	}
+	if config.LogLevel().String() == "unknown" {
+		return fmt.Errorf("loglevel is invalid: %v\n", string(config.LogLevel()))
+	}
+	return nil
+}
+
+// ConnectionString returns a Data Source Name (DSN) to be used by go clients for connecting to a running server.
+func ConnectionString(config ServerConfig) string {
+	return fmt.Sprintf("%v:%v@tcp(%v:%v)/dolt", config.User(), config.Password(), config.Host(), config.Port())
```

Comment:
/dolt is wrong for most configs Either pick a DB that exists, or don't name one.

Reply:
fixed